### ISSUE TITLE
Add conditions in status of volume replication

### ISSUE THREAD

**Issue by iamniting** [csi-addons/volume-replication-operator#48]
As of now, we have only the State field in the status of volume replication. Which does not give us much clarity on what is the current state. Adding conditions in status will give a clear picture of the volume replication state.

**Comment by iamniting** [comment]
We can do something like this. We will mark the Spec.ReplicaState condition True when we met the state and others as false.

```yaml
status:
  conditions:
    - lastTransitionTime: "2019-10-22T16:29:24Z"
      status: "True"
      type: IsPrimary
    - lastTransitionTime: "2019-10-22T16:29:24Z"
      status: "False"
      type: IsSecondary
    - lastTransitionTime: "2019-10-22T16:29:31Z"
      status: "False"
      type: IsResyncing
```

**Comment by iamniting** [comment]
@ShyamsundarR @leseb any thoughts on this.

**Comment by ShyamsundarR** [comment]
Conditions are ["Condition contains details for one aspect of the current state of this API Resource."](https://github.com/kubernetes/kubernetes/blob/eb44008ce2d05adcc9cd9c6a2950a53a2bf3caeb/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1376). Hence, IMO we should not make the conditions the state itself.

Why not go with a status.replicaState and the following status.conditions

status.replicaState: [Primary|Secondary|Unknown]
NOTE: There would be other status variables here (like last sync time maybe), but just putting forth one for related conditions
status.conditions:
  - Completed
    - Reason: [Success] [Error|Progressing|Degraded]
    - Status: [True] [False]
  - Degraded
    - Reason: [Unrechable|Error] [Healthy]
    - Status: [True] [False]
  - Progressing:
    - Reason: [Resyncing|Error] [Completed]
    - Status: [True] [False]

**Comment by iamniting** [comment]
@ShyamsundarR Can you pls elaborate on the `Degraded`

**Comment by ShyamsundarR** [comment]
Here is where I think "Degraded"  maybe useful.

Periodically we may check the image status to update last snapshot sync time etc. During this if either the snapshots are not syncing or other errors are noted, a degraded condition may help.

I also wanted to add a note on what the user expectation could be, here it is,
- User checks status.replicaState and status.conditions.Type[Completed][Status] == True and status.conditions.Type[Completed][ObservedGeneration] == spec.metadata.generation to know that replication is as per the desired state of the current generation and the reconciliation to this state is complete.

**Comment by sp98** [comment]
fixed in #61
